### DISCUSSION TITLE
Reduce avatars

### DISCUSSION THREAD

**Discussion by hirbod** [flyerhq/react-native-chat-ui#43] [Ideas / Feature Requests] [answered]
One feature I liked with react-native-gifted-chat was the ability, to reduce the amount of avatar images. 
Currently, your implementation shows the avatar right next to every bubble. RNGC had the ability to show the avatar only at the last message. When the last message got interrupted with a message by yourself, it would show one image again.

I don't know of this kind of features were the reason for the performance lacks, but if this can be done in a performant, I would love to see it.

![IMG_8028](https://user-images.githubusercontent.com/504909/149160966-ab1e874a-05af-40f4-bdb2-717de21bf7ad.jpg)
![IMG_8029](https://user-images.githubusercontent.com/504909/149160977-989c2d76-628d-471c-af44-8083e69c5bbc.jpg)

**Comment by demchenkoalex** [comment]
Hey @Hirbod this is very strange cause I did not implement that feature to show the avatar on every message. It actually should be shown only on the last one.

Opened the example and it is working at it should work (and as you request). How did you made it work like opposite?

![Simulator Screen Shot - iPhone 13 - 2022-01-17 at 00 18 15](https://user-images.githubusercontent.com/14123304/149682185-c691b91b-d324-418d-a646-55d18e6b69f8.png)

**Comment by hirbod** [comment, marked as answer]
Hi @demchenkoalex,

thanks for your heads up. I'll dig into this once again. I might've messed up with my mock data and user ids. I will recheck to make sure I haven't caused the issue.

**Comment by hirbod** [comment]
Hi @demchenkoalex

haven't been working on the Chat since Januar and just returned to finish it yet. I just setup everything once again, made sure that my Objects and Data are correct, but stumbled again into this issue.

Turned out updatedAt/createdAt was the reason. My server was sending them as ISO Date String, while your lib needs an Unix timestamp. I am converting my data to unix-timestamps now, which fixed the avatar issue.
